ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/PmtTransactionPool.java

```diff
@@ -0,0 +1,113 @@
+package org.hyperledger.besu.ethereum.privacy;
+
+import org.hyperledger.besu.datatypes.Hash;
+import org.hyperledger.besu.ethereum.chain.BlockAddedEvent;
+import org.hyperledger.besu.ethereum.chain.BlockAddedObserver;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class PmtTransactionPool implements BlockAddedObserver {
+  private static final Logger LOG = LogManager.getLogger();
+  private final Map<Hash, PmtTransactionTracker> pmtPool;
+
+  public PmtTransactionPool() {
+    this.pmtPool = new HashMap<>();
+  }
+
+  @Override
+  public void onBlockAdded(final BlockAddedEvent event) {
+    // remove from the map any transactions that went into this block
+    // TODO take into account re-orgs
+    event
+        .getAddedTransactions()
+        .forEach(
+            tx -> {
+              LOG.debug("removing " + pmtPool.containsKey(tx.getHash()));
```

Comment:
For the normal transaction pool, when multiple transactions are submitted with the same *(sender, nonce)* then one will supersede the other depending on their relative timing and gas price. I assume this handled the same way for PMT transactions which--due to zero gas price--would always prefer the first submitted transaction.

Thus when a user submits multiple transactions with the same nonce, all but the committed one would stay in this PmtTransactionPool forever.

Shouldn't we instead delete all transactions from the same sender with nonce `<=` the one that was just committed to chain?

This may complicate reinsertion of transaction during re-orgs.